libs/core/spi.cpp

```diff
@@ -0,0 +1,54 @@
+#include "pxt.h"
+#include "ErrorNo.h"
+
+namespace pins {
+    static codal::SPI *spi = NULL;
+    static void initSPI() {
+        if (NULL == spi)
+            spi = new CODAL_SPI(*LOOKUP_PIN(MOSI), *LOOKUP_PIN(MISO), *LOOKUP_PIN(SCK));
+    }
+
+    /**
+    * Write to the SPI slave and return the response
+    * @param value Data to be sent to the SPI slave
+    */
+    //% help=pins/spi-write weight=5 advanced=true
+    //% blockId=spi_write block="spi write %value"
+    int spiWrite(int value) {
+        initSPI();
+        return spi->write(value);
+    }
+
+    /**
+    * Writes a given command to SPI bus, and afterwards reads the response.
+    */
+    //% help=pins/spi-transfer weight=4 advanced=true
+    //% blockId=spi_transfer block="spi transfer %command into %response"
+    void spiTransfer(Buffer command, Buffer response) {
+        initSPI();
```

Comment:
Allow for command and/or response being null - pass NULL,0 to the codal function then